### ISSUE TITLE
Disabled Grid allows focusing cells using keyboard

### ISSUE THREAD

**Issue by anezthes** [vaadin/flow-components#1831]
### Description

A disabled Grid doesn't allow selecting items using pointing devices, but it is possible to select an item (at least visually) using keyboard.

### Expected outcome

Selection should be fully disabled.

### Actual outcome

Selection is possible using keyboard.

### Minimal reproducible example

```
Grid<Item> grid = new Grid();
grid.addColumn(Item::getName).setHeader("Name");
grid.setEnabled(false);

ArrayList<Item> items = new ArrayList();
for (int i = 0; i < 10; i++) {
  items.add(new Item("Item " + i));
}
grid.setItems(items);

...

private class Item {

  String name;

  Item(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }
}
```

### Environment

- Vaadin 14.6.3 (Flow)
- macOS 11.2.3

### Browsers Affected

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

**Comment by vlukashov** [comment]
NOTE: the `vaadin-grid` web-component does not have a `disabled` mode at all.

**Comment by rolfsmeds** [comment]
Changed the title to refer to cell focus rather than selection, as selection is correctly disabled, but cell focus works normally (both with mouse and keyboard, although only the latter has a visual indication).

Question is: Should cell focus, or indeed all focus, be disabled when the Grid is disabled?

**Comment by tomivirkki** [comment]
Grid shouldn't have `setEnabled` API in the first place, because `<vaadin-grid>` doesn't have `disabled` mode/feature either (as stated above).

I'd be in favor of first designing how `<vaadin-grid disabled>` should work in general and then implementing it in the Web Component as a feature. Once the feature is in place, the `disabled` mode should be taken into account whenever a new feature gets added to `<vaadin-grid>` to avoid inconsistent behavior, like the one in question, between the Flow and Web components.

In case `disabled` doesn't make sense for `<vaadin-grid>`, the erroneous `setEnabled` API could just be removed from the Flow Grid.

**Comment by DiegoCardoso** [comment]
> In case disabled doesn't make sense for <vaadin-grid>, the erroneous setEnabled API could just be removed from the Flow Grid.

I guess disabled makes sense to Flow Grid because of the Editor:

https://user-images.githubusercontent.com/262432/123935195-ac133e80-d99c-11eb-8ed7-ac049ac59b71.mp4

**Comment by tomivirkki** [comment]
If it's only required for the editor (another Flow-only feature), there could be a dedicated API for disabling/enabling the editor.

**Comment by rolfsmeds** [comment]
Created a [separate ticket](https://github.com/vaadin/web-components/issues/2170) for implementing a proper disabled state in vaadin-grid. Closing this in favor of that.
